gsa/src/web/pages/tasks/listpage.js

```diff
@@ -156,6 +166,25 @@ const TasksListPage = () => {
     [deleteTask, refetch, showError],
   );
 
+  useEffect(() => {
+    // load all filtered tasks if selection is all filtered
+    if (selectionType === SelectionType.SELECTION_FILTER) {
+      getAllFiltered({filterString: filter.all().toFilterString()});
+    }
+  }, [selectionType, filter, getAllFiltered]);
+
+  const handleBulkDeleteTask = () => {
+    let tasksToDelete;
+    if (selectionType === SelectionType.SELECTION_USER) {
+      tasksToDelete = getEntityIds(selected);
+    } else if (selectionType === SelectionType.SELECTION_PAGE_CONTENTS) {
+      tasksToDelete = getEntityIds(tasks);
+    } else {
+      tasksToDelete = getEntityIds(allFilteredTasks);
```

Comment:
Much better already. But this could result in a race condition when loading of all filtered tasks takes a while. In that case getEntityIds would be called with an empty or old allFilteredTasks variable. I see two possible solutions:

1. query getAllFiltered when clicking on the bulk delete and using the returned promise
2. implement something like deleteAllTasksByFilter in the backend